google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicSearchAds.java

```diff
@@ -0,0 +1,335 @@
+import com.google.ads.googleads.examples.utils.CodeSampleParams;
+import com.google.ads.googleads.lib.GoogleAdsClient;
+import com.google.ads.googleads.v1.common.ExpandedDynamicSearchAdInfo;
+import com.google.ads.googleads.v1.common.ManualCpc;
+import com.google.ads.googleads.v1.common.WebpageConditionInfo;
+import com.google.ads.googleads.v1.common.WebpageInfo;
+import com.google.ads.googleads.v1.enums.AdGroupAdStatusEnum.AdGroupAdStatus;
+import com.google.ads.googleads.v1.enums.AdGroupCriterionStatusEnum.AdGroupCriterionStatus;
+import com.google.ads.googleads.v1.enums.AdGroupStatusEnum.AdGroupStatus;
+import com.google.ads.googleads.v1.enums.AdGroupTypeEnum.AdGroupType;
+import com.google.ads.googleads.v1.enums.AdvertisingChannelTypeEnum.AdvertisingChannelType;
+import com.google.ads.googleads.v1.enums.BudgetDeliveryMethodEnum.BudgetDeliveryMethod;
+import com.google.ads.googleads.v1.enums.CampaignStatusEnum.CampaignStatus;
+import com.google.ads.googleads.v1.enums.WebpageConditionOperandEnum.WebpageConditionOperand;
+import com.google.ads.googleads.v1.errors.GoogleAdsError;
+import com.google.ads.googleads.v1.errors.GoogleAdsException;
+import com.google.ads.googleads.v1.resources.Ad;
+import com.google.ads.googleads.v1.resources.AdGroup;
+import com.google.ads.googleads.v1.resources.AdGroupAd;
+import com.google.ads.googleads.v1.resources.AdGroupCriterion;
+import com.google.ads.googleads.v1.resources.Campaign;
+import com.google.ads.googleads.v1.resources.Campaign.DynamicSearchAdsSetting;
+import com.google.ads.googleads.v1.resources.CampaignBudget;
+import com.google.ads.googleads.v1.services.AdGroupAdOperation;
+import com.google.ads.googleads.v1.services.AdGroupAdServiceClient;
+import com.google.ads.googleads.v1.services.AdGroupCriterionOperation;
+import com.google.ads.googleads.v1.services.AdGroupCriterionServiceClient;
+import com.google.ads.googleads.v1.services.AdGroupOperation;
+import com.google.ads.googleads.v1.services.AdGroupServiceClient;
+import com.google.ads.googleads.v1.services.CampaignBudgetOperation;
+import com.google.ads.googleads.v1.services.CampaignBudgetServiceClient;
+import com.google.ads.googleads.v1.services.CampaignOperation;
+import com.google.ads.googleads.v1.services.CampaignServiceClient;
+import com.google.ads.googleads.v1.services.MutateAdGroupAdsResponse;
+import com.google.ads.googleads.v1.services.MutateAdGroupCriteriaResponse;
+import com.google.ads.googleads.v1.services.MutateAdGroupsResponse;
+import com.google.ads.googleads.v1.services.MutateCampaignBudgetsResponse;
+import com.google.ads.googleads.v1.services.MutateCampaignsResponse;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.Int64Value;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import org.joda.time.DateTime;
+
+/** This example adds a new dynamic search ad (DSA) and webpage targeting criteria for the DSA. */
```

Comment:
Nit: start comments so they could complete the sentence "This thing does..."

This comment can start "Adds a new DSA..."

Reply:
Friendly ping on this one remaining comment.

Reply:
Thanks, I missed that one. Just fixed.